cowbird/permissions_synchronizer.py

```diff
@@ -74,59 +74,178 @@ def __init__(self,
-                        for mapping_pt in mapping]
+        self.resource_keys = {res_key: res for svc in self.services.values() for res_key, res in svc.items()}
+        self.mapping = [{res_key: perms for res_key, perms in mapping_pt.items()
+                         if res_key in self.resource_keys.keys()} for mapping_pt in mapping]
 
-    def resource_match(self, permission):
-        # type: (Permission) -> bool
-        """
-        Define if the permission name is covered by this sync point.
-        """
-        return permission.resource_full_name.startswith(self.services[permission.service_name])
-
-    def find_match(self, permission):
-        # type: (Permission) -> Generator[Tuple[str, str], None, None]
+    def find_permissions_to_sync(self, permission, res_root_key):
+        # type: (Permission, str) -> Generator[Tuple[str, str], None, None]
         """
         Search and yield for every match a (service, permission name) tuple that is mapped with this permission.
         """
-        # check if the permission name is covered by this sync point
-        if not self.resource_match(permission):
-            return
-
         # For each permission mapping
         for mapping in self.mapping:
             # Does the current service has some mapped permissions?
-            if permission.service_name not in mapping:
+            if res_root_key not in mapping:
                 continue
             # Does the current permission access is covered?
-            if permission.name not in mapping[permission.service_name]:
+            if permission.name not in mapping[res_root_key]:
                 continue
             # This permission is mapped : yields matches
-            for svc, mapped_perm_name in mapping.items():
+            for res, mapped_perm_name in mapping.items():
                 # Don't synch with itself
-                if svc == permission.service_name:
+                if res == res_root_key:
                     continue
                 for perm_name in mapped_perm_name:
-                    yield svc, perm_name
+                    yield res, perm_name
 
-    def sync(self, perm_operation, permission):
-        # type: (Callable[Permission], Permission) -> None
+    def _find_matching_res(self, service_name, resource_nametype_path):
+        # type: (str, str) -> (str, int)
+        """
+        Finds a resource key that matches the input resource path, in the sync_permissions config.
+        Note that it returns the longest match and only the named segments of the path are included in the length value.
+        Any tokenized segment is ignored in the length.
+
+        @param service_name Name of the service associated with the input resource.
+        @param resource_nametype_path Full resource path name, which includes the type of each segment
+                                       (ex.: /name1:type1/name2:type2)
```

Comment:
Mmhm, it seems not to be consistent throughout the repo already, in the older files. I will have to go over all those files, I guess we prefer the `:param <name>:` notation since it is the rst format used by sphinx.